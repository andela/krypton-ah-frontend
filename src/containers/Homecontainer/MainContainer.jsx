/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Carousel from '../../components/carousel/index';
import Sidewidget from '../sideWidgetContainer';
import FeaturedArticles from './featuredArticles';
import PopularArticles from './popular';
import './HomeContainer.scss';
import {
  fetchpopular,
  fetchfeatured
} from '../../actions/fetchArticlesAction/fetchArticlesActions';
import Title from '../../components/sectionHeader';
import loader from '../loaders/responsechecker';

class MainContainer extends Component {
  componentWillMount() {
    const { fetchpopular, fetchfeatured } = this.props;
    fetchpopular();
    fetchfeatured();
  }

  render() {
    const { featuredArticles, popularArticles } = this.props;
    return (
      <div className="main">
        <Grid className="content" divided="vertically">
          <Grid.Row columns={1}>
            {loader(featuredArticles.featuredArticlesfailure,
              <Carousel featured={featuredArticles.featuredArticlesResponse} />)}
            <Grid.Column className="second_grid">
              <Title text="featured" />
              {loader(featuredArticles.featuredArticlesfailure,
                <FeaturedArticles articles={featuredArticles.featuredArticlesResponse} />)}
              <Title text="Popular" />
              {loader(popularArticles.popularArticlesResponsefailure,
                <PopularArticles popular={popularArticles.popularArticlesResponse} />)}
            </Grid.Column>
          </Grid.Row>
        </Grid>
        <Sidewidget />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  featuredArticles: state.FeaturedArticlesReducer,
  popularArticles: state.popularArticlesReducer
});

export default connect(
  mapStateToProps,
  { fetchpopular, fetchfeatured }
)(MainContainer);

MainContainer.defaultProps = {
  featuredArticles: [],
  fetchpopular: [],
  fetchfeatured: [],
  popularArticles: []
};

MainContainer.propTypes = {
  popularArticles: PropTypes.array,
  featuredArticles: PropTypes.array,
  fetchpopular: PropTypes.func,
  fetchfeatured: PropTypes.func
};
