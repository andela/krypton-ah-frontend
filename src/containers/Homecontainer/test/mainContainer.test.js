import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from '../MainContainer';
import PopularArticles from '../popular';
import FeaturedArticles from '../featuredArticles';
import Loader from '../../loaders/componentLoader';

describe('home container test', () => {
  it('should test for the home component', () => {
    const wrapper = shallow(<MainContainer />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the popular articles component', () => {
    const wrapper = shallow(<PopularArticles />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the featuredarticles component', () => {
    const wrapper = shallow(<FeaturedArticles />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the featuredarticles component', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
