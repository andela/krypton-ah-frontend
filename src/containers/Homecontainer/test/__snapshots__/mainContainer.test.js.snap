// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home container test should test for the featuredarticles component 1`] = `
<Fragment>
  <Grid
    columns={2}
    divided={true}
  >
    <GridRow>
      <GridColumn
        className="rightGrid"
      >
        <Link
          replace={false}
          to="/article/65719288-0395-445e-b587-2b98b70bdec9"
        >
          <Card
            header="This is a post title"
            image="https://farm4.staticflickr.com/3894/15008518202_c265dfa55f_h.jpg"
          />
        </Link>
        <Card
          className="description"
        >
          <p
            className="descriptionBody"
          >
            Descriptive title
          </p>
          <Connect(Icons)
            date="2019-01-23T07:09:10.447Z"
            disabled={true}
            id="65719288-0395-445e-b587-2b98b70bdec9"
            name="hakeem ibrahim"
            rating={3}
            ratingIcon={null}
          />
        </Card>
      </GridColumn>
      <GridColumn>
        <ArticlesFeedContainer
          articles={
            Array [
              Object {
                "articleAuthor": Object {
                  "firstname": "hakeem",
                  "lastname": "ibrahim",
                  "userprofile": null,
                },
                "authorId": "65719288-0395-445e-b587-2b98b70bdec9",
                "averageRating": 3,
                "content": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                "createdAt": "2019-01-23T07:09:10.447Z",
                "data": Object {
                  "message": Object {
                    "id": Array [
                      "id must be a valid uuid",
                    ],
                  },
                  "success": true,
                },
                "description": "Descriptive title",
                "featuredImageUrl": "https://farm4.staticflickr.com/3894/15008518202_c265dfa55f_h.jpg",
                "id": "65719288-0395-445e-b587-2b98b70bdec9",
                "isPublished": true,
                "numberOfReviews": null,
                "readTime": 5,
                "slug": "this-is-a-post-title-l78hgybf",
                "tags": Array [],
                "title": "This is a post title",
                "updatedAt": "2019-01-23T07:09:10.447Z",
              },
            ]
          }
          className="featured-widget"
        />
      </GridColumn>
    </GridRow>
  </Grid>
</Fragment>
`;

exports[`home container test should test for the featuredarticles component 2`] = `<componentloader />`;

exports[`home container test should test for the featuredarticles component 3`] = `
<Segment>
  <Dimmer
    active={true}
    inverted={true}
  >
    <Loader
      disabled={true}
      size="medium"
    />
  </Dimmer>
  <Image
    as="img"
    src="https://react.semantic-ui.com/images/wireframe/paragraph.png"
    ui={true}
  />
</Segment>
`;

exports[`home container test should test for the home component 1`] = `
<div
  className="main"
>
  <Grid
    className="content"
    divided="vertically"
  >
    <GridRow
      columns={1}
    >
      <carousel
        featured={
          Object {
            "data": Array [
              Object {
                "articleAuthor": Object {
                  "firstname": "hakeem",
                  "lastname": "ibrahim",
                  "userprofile": null,
                },
                "authorId": "65719288-0395-445e-b587-2b98b70bdec9",
                "averageRating": 3,
                "content": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                "createdAt": "2019-01-23T07:09:10.447Z",
                "data": Object {
                  "message": Object {
                    "id": Array [
                      "id must be a valid uuid",
                    ],
                  },
                  "success": true,
                },
                "description": "Descriptive title",
                "featuredImageUrl": "https://farm4.staticflickr.com/3894/15008518202_c265dfa55f_h.jpg",
                "id": "65719288-0395-445e-b587-2b98b70bdec9",
                "isPublished": true,
                "numberOfReviews": null,
                "readTime": 5,
                "slug": "this-is-a-post-title-l78hgybf",
                "tags": Array [],
                "title": "This is a post title",
                "updatedAt": "2019-01-23T07:09:10.447Z",
              },
            ],
          }
        }
      />
      <GridColumn
        className="second_grid"
      >
        <sectionHeader
          text="featured"
        />
        <FeaturedArticles
          articles={Array []}
        />
        <sectionHeader
          text="Popular"
        />
        <PopularArticles
          popular={Array []}
        />
      </GridColumn>
    </GridRow>
  </Grid>
  <Connect(WidgetContainer) />
</div>
`;

exports[`home container test should test for the popular articles component 1`] = `<componentloader />`;

exports[`home container test should test for the popular articles component 2`] = `
<Fragment>
  <Grid
    columns={2}
    divided={true}
  >
    <GridRow>
      <Grid
        className="reveal"
        columns={2}
        container={true}
      >
        <GridColumn
          key="65719288-0395-445e-b587-2b98b70bdec9"
        >
          <Card
            className="description"
          >
            <Link
              replace={false}
              to="/article/65719288-0395-445e-b587-2b98b70bdec9"
            >
              <p
                className="popular title"
              >
                This is a post title
              </p>
            </Link>
            <p
              className="revealDescription"
            >
              Descriptive title
            </p>
          </Card>
          <Image
            as="img"
            src="https://farm4.staticflickr.com/3894/15008518202_c265dfa55f_h.jpg"
            ui={true}
          />
        </GridColumn>
      </Grid>
    </GridRow>
  </Grid>
</Fragment>
`;
